src/Snt/Capi/Repository/Article/FindBySectionParameters.php

```diff
@@ -0,0 +1,217 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildApiHttpPathAndQuery()
+    {
+        $path = sprintf(
+            self::URL_PATTERN,
+            $this->publicationId,
+            implode(',', $this->sections),
+            http_build_query($this->parameters)
+        );
+
+        $path = rtrim($path, '?');
+
+        return ApiHttpPathAndQuery::createForPath($path);
+    }
+
+    /**
+     * @param string $view
+     *
+     * @return $this
+     * @throws InvalidArgumentException
+     */
+    public function setView($view)
+    {
+        if (!in_array($view, ['min', 'teaser'])) {
+            throw new InvalidArgumentException(
+                sprintf('Value "%s" is not allowed. Allowed are "min", "teaser"', $view)
+            );
+        }
+
+        $this->parameters['view'] = $view;
+
+        return $this;
+    }
+
+    /**
+     * @param int $offset
+     *
+     * @return $this
+     */
+    public function setOffset($offset)
+    {
+        $this->parameters['offset'] = (int) $offset;
+
+        return $this;
+    }
+
+    /**
+     * @param int $limit
+     *
+     * @return $this
+     */
+    public function setLimit($limit)
+    {
+        $this->parameters['limit'] = (int) $limit;
+
+        return $this;
+    }
+
+    /**
+     * @param string $since
+     *
+     * @return $this
+     * @throws InvalidArgumentException
+     */
+    public function setSince($since)
+    {
+        try {
+            $date = new DateTime($since);
+            $this->parameters['since'] = $date->format('YYYY-mm-dd');
+        } catch (Exception $e) {
+            throw new InvalidArgumentException(sprintf('Argument "%s" is not a valid date', $since), 0, $e);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param string $until
+     *
+     * @return $this
+     * @throws InvalidArgumentException
+     */
+    public function setUntil($until)
+    {
+        try {
+            $date = new DateTime($until);
+            $this->parameters['until'] = $date->format('YYYY-mm-dd');
+        } catch (Exception $e) {
+            throw new InvalidArgumentException(sprintf('Argument "%s" is not a valid date', $until), 0, $e);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param int $hotnessFrom
+     *
+     * @return $this
+     * @throw InvalidArgumentException
+     */
+    public function setHotnessFrom($hotnessFrom)
+    {
+        $hotnessFrom = (int) $hotnessFrom;
+        if ($hotnessFrom < 0 || $hotnessFrom > 100) {
+            throw new InvalidArgumentException(sprintf('Argument "%s" should be from [0; 100] range', $hotnessFrom));
+        }
+
+        $this->parameters['hotnessFrom'] = $hotnessFrom;
+
+        return $this;
+    }
+
+    /**
+     * @param int $hotnessTo
+     *
+     * @return $this
+     * @throw InvalidArgumentException
+     */
+    public function setHotnessTo($hotnessTo)
+    {
+        $hotnessTo = (int) $hotnessTo;
+        if ($hotnessTo < 0 || $hotnessTo > 100) {
```

Comment:
four times same condition - maybe private method `isNotBetweenZeroAndOneHundred($value)` or `valueIsNotBetween($value, $from, $to)` what do you think?